projects/HVAC/vent_control/vent.ino

```diff
@@ -0,0 +1,260 @@
+/*
+
+motor: 28BYJ48-5V
+  steps: 8
+  stride angle: 5.625/64
+
+  loops to 1 revolution: 360/(8 * 5.625/64) = 512
+
+            | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 |
+  4 orange  | x | x |   |   |   |   |   | x |
+  ----------|---|---|---|---|---|---|---|---|
+  3 yellow  |   | x | x | x |   |   |   |   |
+  ----------|---|---|---|---|---|---|---|---|
+  2 pink    |   |   |   | x | x | x |   |   |
+  ----------|---|---|---|---|---|---|---|---|
+  1 blue    |   |   |   |   |   | x | x | x |
+  ----------|---|---|---|---|---|---|---|---|
+
+Example:
+
+@VENT01>set degrees: 90
+@VENT01>set percents: 25
+
+*/
+
+#include <SyrotaAutomation1.h>
+
+void motor_step_cw();
+void motor_step_ccw();
+bool motor_zero();
+void motor_rotate_degrees(int degrees);
+void motor_rotate_percents(int percents);
+void motor_rotate_to(int position);
+
+void set_error(char *message);
+
+// Handles:
+//   * set degrees: -360..360
+//   * set percents: -100..100
+void process_command();
+
+int motor_position = 0;
+// steps per rotation
+const int motor_steps_per_rotation = 512;
+const float motor_steps_per_degree = motor_steps_per_rotation / 360.0;
+
+const int motor_sensor_pin = 2;
+const int blink_pin = 13;
+
+const int serial_pin = 3;
+SyrotaAutomation net = SyrotaAutomation(serial_pin);
+const char* net_address = "VENT01";
+
+
+void setup() {
+  // Serial
+  Serial.begin(9600);
+
+  // Error light
+  pinMode(blink_pin, OUTPUT);
+  digitalWrite(blink_pin, LOW);
+
+  // Motor pins
+  pinMode(8,  OUTPUT);
+  pinMode(9,  OUTPUT);
+  pinMode(10, OUTPUT);
+  pinMode(11, OUTPUT);
+
+  pinMode(motor_sensor_pin, INPUT);
+
+  if(!motor_zero()) {
+    set_error("Cannot zero motor!");
+  }
+
+  strcpy(net.deviceID, net_address);
+
+  blink_happy();
+  net.sendResponse("READY");
+
+}
+
+void loop() {
+  // Example motor handling
+  // delay(1000);
+  // motor_rotate_to(128);
+  // motor_rotate_degrees(90);
+  // motor_rotate_percents(25);
+  // delay(1000);
+  // motor_rotate_to(0);
+  // motor_rotate_degrees(0);
+  // motor_rotate_percents(0);
+  // delay(1000);
+
+  if(net.messageReceived()) {
+    process_command();
+  }
+}
+
+// blink successfull init
+void blink_happy() {
+  for(int i = 0; i < 5; i++) {
+    digitalWrite(blink_pin, HIGH);
+    delay(100);
+    digitalWrite(blink_pin, LOW);
+    delay(50);
+  }
+}
+
+// -----------
+
+void process_command() {
+  char buffer [100];
+  long int value = 0;
+
+  if(net.assertCommandStarts("set degrees:", buffer)) {
+    value = strtol(buffer, NULL, 10);
+    motor_rotate_degrees(value);
```

Comment:
command suggest you'd set damper to certain angle. But looks like this function rotates it +N degrees. In other words if I call "set degrees:10" twice, I expect second one will do nothing, as it's already at 10 degrees. But it will in fact do +10deg, then another +10deg.